pyemma/util/tests/test_discrete_trajectories.py

```diff
@@ -199,5 +196,116 @@ def test_sample_by_state_replace_subset(self):
+        core_set = np.arange(2, 7)
+        dtrajs_core, offsets, n_cores = dt.milestone_counting(dtrajs, core_set)
+        self.assertEqual(n_cores, 5)
+        np.testing.assert_equal(dtrajs_core, expected)
+
+    def test_core_sets_2(self):
+        dtrajs = [np.array([0, 0, 2, 1, 2])]
+        expected = [np.array([2, 1, 2])]
+        dtrajs_core, offsets, n_cores = dt.milestone_counting(dtrajs, core_set=np.arange(1, 3))
+        self.assertEqual(n_cores, 2)
+        np.testing.assert_equal(dtrajs_core, expected)
+
+    def test_core_sets_3(self):
+        dtrajs = [np.array([2, 0, 1, 1, 2])]
+        expected = [np.array([2, 2, 1, 1, 2])]
+        dtrajs_core, offsets, n_cores = dt.milestone_counting(dtrajs, core_set=np.arange(1, 3))
+        self.assertEqual(n_cores, 2)
+        np.testing.assert_equal(dtrajs_core, expected)
+
+    def test_core_sets_4(self):
+        dtrajs = [np.array([2, 0, 0, 2, 0, 2, 0, 2])]
+        dtrajs_core, offsets, n_cores = dt.milestone_counting(dtrajs, core_set=[1, 2])
+        self.assertEqual(n_cores, 2)
+        np.testing.assert_equal(dtrajs_core, [np.ones_like(dtrajs[0]) * 2])
+
+    def test_core_sets_5(self):
+        dtrajs = [np.array([2, 2, 2, 2, 2, 2, 2, 0])]
+        dtrajs_core, offsets, n_cores = dt.milestone_counting(dtrajs, core_set=[2])
+        self.assertEqual(n_cores, 1)
+        np.testing.assert_equal(dtrajs_core, [np.ones_like(dtrajs[0]) * 2])
+
+    def test_core_set_all_non_negative(self):
+        dtrajs = [np.array([-1, 2, -1, 0, 0, 1, -1, 0, 1, 0, 2, -1])]
+        expected =   [np.array([2,  2, 0, 0, 1,  1, 0, 1, 0, 2, 2])]
+        dtrajs_core, offsets, n_cores = dt.milestone_counting(dtrajs, core_set=None)
+        np.testing.assert_equal(offsets, [1])
+        np.testing.assert_equal(n_cores, 3)
+        np.testing.assert_equal(dtrajs_core, expected)
+
+    def test_core_sets_6(self):
+        dtrajs = [np.array([0, 1, 1, 2]), np.array([0, 0, 0])]
+        import warnings
+
+        if sys.version_info[0] == 2: # yeah python 2 bugs ftw...
+            if hasattr(dt.milestone_counting, '__globals__'):
+                if dt.milestone_counting.__globals__.has_key('__warningregistry__'):
+                    dt.milestone_counting.__globals__['__warningregistry__'].clear()
+
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always", category=UserWarning, append=False)
+            dtraj_core, offsets, _ = dt.milestone_counting(dtrajs, core_set=[1, 2])
+            assert len(w) == 2
+            assert all(issubclass(x.category, UserWarning) for x in w)
+            assert "had to be truncated" in str(w[0].message)
+            assert "never visited a core set" in str(w[1].message)
+
+    def test_realistic_random(self):
+        n_states = 10
+        n_traj = 10
+        dtrajs = [np.random.randint(0, n_states, size=1000) for _ in range(n_traj)]
+        core_set = np.random.randint(0, n_states, size=30)
```

Comment:
30 states out of 10 states will be assigned core sets? does that make sense?

Reply:
btw, how are redundant states handled, i.e. if user passes `core_set = [0, 0, 1]`?